fix: transition from allow to expect for turning off clippy lints

RaghavArora14 opened the pull request:
/claim #545
This PR addresses the transition from `#[allow(clippy::...)]` to `#[expect(clippy::...)]` as described in issue #545.

### Changes Made

#### Configuration Update:
- Modified the `Cargo.toml` file:
  - Added `allow_attributes = ["warn"]` under `[workspace.lints.clippy]` to enable warnings for remaining `#[allow(...)]` attributes.

#### Systematic Codebase Updates:
- Used a regular expression (`#\[allow\(clippy::[^)]*\)`) to locate all `#[allow(clippy::...)]` occurrences.
- Replaced `#[allow(clippy::...)]` with `#[expect(clippy::...)]`, preserving the original lint names.
- Performed manual reviews for each change to ensure correctness.

#### Incremental Commits:
- Changes were committed incrementally for clear history and easy debugging.

iajoiner reviewed (commented):
@RaghavArora14 Really thanks for your work! Please address one comment I left and do the following since we have a mandatory regex-based check on commit messages.
```bash
git reset --soft HEAD~4
git add --all
git commit -m 'fix: transition from allow to expect for turning off clippy lints'
git push -f origin
```

Thanks again!

RaghavArora14 commented on Cargo.toml:
```diff
@@ -9,6 +9,7 @@ repository = "https://github.com/spaceandtimelabs/sxt-proof-of-sql"
 version = "0.0.0" # DO NOT CHANGE THIS LINE! This will be automatically updated
 license-file = "LICENSE"
 
+
```
So I just added a space as I didn't previously have a properly formated commit message before so just added a space and pushed that commit so that I can push the commit message with the right format. I just got a lil confused

RaghavArora14 commented:
@iajoiner Hi I have addressed the comment and pushed as per requested for the regex-based check on commit messages.
Please review it once and let me know if anything further is required

iajoiner commented:
@RaghavArora14 Sorry you didn't. Well I can clean it up for you.

JayWhite2357 commented:
@RaghavArora14 
Thanks for the PR!
Take a look at the CI jobs. Several are not passing. When you are able to get them to pass, we'll take a close look at the PR.
I would recommend running the jona locally so you don't have to wait for us to trigger the CI.
